src/main/java/com/hollingsworth/arsnouveau/api/item/IWandable.java

```diff
@@ -3,34 +3,110 @@
 import com.hollingsworth.arsnouveau.client.particle.ColorPos;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Direction;
+import net.minecraft.core.GlobalPos;
+import net.minecraft.resources.ResourceKey;
 import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.Level;
 import org.jetbrains.annotations.Nullable;
 
+import java.awt.*;
 import java.util.List;
 
 public interface IWandable {
     /**
      * When the wand has made 2 connections, block -> block, block -> entity, entity -> block, or entity -> entity.
      * The FIRST IWandable in the chain is called.
      */
+    default void onFinishedConnectionFirst(@Nullable GlobalPos storedPos, @Nullable LivingEntity storedEntity, Player playerEntity) {
+        ResourceKey<Level> dim;
+        if (storedPos != null) {
+            dim = storedPos.dimension();
+        } else if (storedEntity != null) {
+            dim = storedEntity.level.dimension();
+        } else {
+            return;
+        }
```

Comment:
Could this be moved into a private function on the interface? Saves repeating the same logic 4 times

Reply:
Resolved